tests/apache/hive/sensors/test_named_hive_partition.py

```diff
@@ -21,68 +21,65 @@ def context():
-        task_id="task-id",
-        partition_names=TEST_PARTITION,
-        metastore_conn_id=TEST_METASTORE_CONN_ID,
-    )
-    with pytest.raises(TaskDeferred) as exc:
-        task.execute(context)
-    assert isinstance(
-        exc.value.trigger, NamedHivePartitionTrigger
-    ), "Trigger is not a NamedHivePartitionTrigger"
-
-
-def test_named_hive_partition_sensor_async_exception():
-    """
-    Asserts that a task is deferred and a NamedHivePartitionTrigger will be fired
-    when the NamedHivePartitionSensorAsync is executed.
-    """
-    task = NamedHivePartitionSensorAsync(
-        task_id="task-id",
-        partition_names=[],
-        metastore_conn_id=TEST_METASTORE_CONN_ID,
-    )
-    with pytest.raises(ValueError):
-        task.execute(context)
-
-
-def test_named_hive_partition_sensor_async_execute_failure(context):
-    """Tests that an AirflowException is raised in case of error event"""
-    task = NamedHivePartitionSensorAsync(
-        task_id="task-id",
-        partition_names=[],
-        metastore_conn_id=TEST_METASTORE_CONN_ID,
-    )
-    with pytest.raises(AirflowException):
-        task.execute_complete(context=None, event={"status": "error", "message": "test failure message"})
+class TestNamedHivePartitionSensorAsync:
+    def test_named_hive_partition_sensor_async(self):
+        """
+        Asserts that a task is deferred and a NamedHivePartitionTrigger will be fired
+        when the NamedHivePartitionSensorAsync is executed.
+        """
+        task = NamedHivePartitionSensorAsync(
+            task_id="task-id",
+            partition_names=TEST_PARTITION,
+            metastore_conn_id=TEST_METASTORE_CONN_ID,
+        )
+        with pytest.raises(TaskDeferred) as exc:
+            task.execute(context)
+        assert isinstance(
+            exc.value.trigger, NamedHivePartitionTrigger
+        ), "Trigger is not a NamedHivePartitionTrigger"
 
+    def test_named_hive_partition_sensor_async_exception(self):
+        """
+        Asserts that a task is deferred and a NamedHivePartitionTrigger will be fired
+        when the NamedHivePartitionSensorAsync is executed.
+        """
+        task = NamedHivePartitionSensorAsync(
+            task_id="task-id",
+            partition_names=[],
+            metastore_conn_id=TEST_METASTORE_CONN_ID,
+        )
+        with pytest.raises(ValueError):
+            task.execute(context)
 
-def test_named_hive_partition_sensor_async_execute_complete():
-    """Asserts that logging occurs as expected"""
-    task = NamedHivePartitionSensorAsync(
-        task_id="task-id",
-        partition_names=TEST_PARTITION,
-        metastore_conn_id=TEST_METASTORE_CONN_ID,
-    )
-    with mock.patch.object(task.log, "info") as mock_log_info:
-        task.execute_complete(
-            context=None, event={"status": "success", "message": "Named hive partition found"}
+    def test_named_hive_partition_sensor_async_execute_failure(self, context):
```

Comment:
is context used from utils are we creating in same tests ?

Reply:
I guess context is a py fixture created in utils

Reply:
this is coming from conftest.py